packages/fxa-settings/src/pages/Signin/SigninRecoveryChoice/container.tsx

```diff
@@ -0,0 +1,127 @@
+import LoadingSpinner from 'fxa-react/components/LoadingSpinner';
+import {
+  AuthUiErrorNos,
+  AuthUiErrors,
+} from '../../../lib/auth-errors/auth-errors';
+
+export const SigninRecoveryChoiceContainer = (_: RouteComponentProps) => {
+  const authClient = useAuthClient();
+  const location = useLocation() as ReturnType<typeof useLocation> & {
+    state: SigninLocationState;
+  };
+  const navigateWithQuery = useNavigateWithQuery();
+  const signinState = getSigninState(location.state);
+
+  const [numBackupCodes, setNumBackupCodes] = useState<number>(0);
+  const [lastFourPhoneDigits, setLastFourPhoneDigits] = useState<string>('');
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    if (!signinState || !signinState.sessionToken) {
+      navigateWithQuery('/signin');
+      return;
+    }
+
+    const fetchData = async () => {
+      try {
+        const { count } = await authClient.getRecoveryCodesExist(
+          signinState.sessionToken
+        );
+        count && setNumBackupCodes(count);
+
+        const { phoneNumber } = await authClient.recoveryPhoneGet(
+          signinState.sessionToken
+        );
+        // TODO verify that recoveryPhoneGet returns a masked phone number (last four digits only)
+        phoneNumber && setLastFourPhoneDigits(phoneNumber.slice(-4));
+
+        // whether or not the user has backup authentication codes,
+        // go directly to the backup authentication codes page if they don't have a phone number
+        // do not render the choice screen
+        if (!phoneNumber) {
+          navigateWithQuery('/signin_recovery_code', {
+            state: { signinState },
+            // ensure back button on signin_recovery_code page skips choice page and returns to signin_totp_code
+            replace: true,
+          });
+          return;
+        }
+
+        if (phoneNumber && (!count || count === 0)) {
+          navigateWithQuery('/signin_recovery_phone', {
+            state: { signinState, lastFourPhoneDigits },
+            // ensure back button on signin_recovery_code page skips choice page and returns to signin_totp_code
+            replace: true,
+          });
+        }
+        return;
+      } catch (err) {
+        if (err.errno === AuthUiErrors.INVALID_TOKEN.errno) {
+          navigateWithQuery('/signin');
+          return;
+        }
+        // if there was another error fetching available recovery methods, go to backup authentication codes page
+        navigateWithQuery('/signin_recovery_code', {
+          state: { signinState },
+          // ensure back button on signin_recovery_code page skips choice page and returns to signin_totp_code
+          replace: true,
+        });
+        return;
+      } finally {
+        setLoading(false);
+      }
+    };
+    fetchData();
+  }, [authClient, lastFourPhoneDigits, signinState, navigateWithQuery]);
+
+  const handlePhoneChoice = async () => {
+    if (!signinState) {
```

Comment:
If you want to use the session token from signin state, you could curry this function with it so you don't have to recheck `if (!signinState) {`, but maybe later we should just pull from local storage instead and curry that.

Reply:
I added a note in `FXA-9875` about reviewing what's included in signinState (vs local storage)